contracts/extensions/nft/NFT.sol

```diff
@@ -0,0 +1,226 @@
+     * @param escrowAddr The address of the escrow adapter, usually it is the address of the TributeNFT adapter.
+     * @param nftAddr The NFT address that must be in ERC721 and allowed/supported by the extension.
+     * @param nftTokenId The NFT token id.
+     */
+    function transferFrom(
+        address escrowAddr,
+        address nftAddr,
+        uint256 nftTokenId
+    ) public hasExtensionAccess(this, AclFlag.TRANSFER_NFT) {
+        require(isNFTAllowed(nftAddr), "non allowed nft");
+        IERC721 erc721 = IERC721(nftAddr);
+        // Move the NFT to the contract address
+        erc721.safeTransferFrom(escrowAddr, address(this), nftTokenId);
+        // Save the asset in the GUILD collection
+        _nftCollection[GUILD][nftAddr].add(nftTokenId);
+        // Keep track of the collected assets
+        _collectedNFTs.add(nftAddr);
+    }
+
+    /**
+     * @notice Ttransfers the NFT token from the extension address to the new owner.
+     * @notice It also updates the internal state to keep track of the all the NFTs collected by the extension.
+     * @notice The caller must have the ACL Flag: RETURN_NFT
+     * @dev Reverts if the NFT is not support/allowed, or is not in ERC721 standard.
+     * @param newOwner The address of the new owner.
+     * @param nftAddr The NFT address that must be in ERC721 and allowed/supported by the extension.
+     * @param nftTokenId The NFT token id.
+     */
+    function returnNFT(
+        address newOwner,
+        address nftAddr,
+        uint256 nftTokenId
+    ) public hasExtensionAccess(this, AclFlag.RETURN_NFT) {
+        require(isNFTAllowed(nftAddr), "nft not allowed");
+
+        // Remove the NFT from the contract address to the actual owner
+        IERC721 erc721 = IERC721(nftAddr);
+        erc721.approve(newOwner, nftTokenId);
+        erc721.safeTransferFrom(address(this), newOwner, nftTokenId);
+        // Remove the asset from the GUILD collection
+        _nftCollection[GUILD][nftAddr].remove(nftTokenId);
+        // If we dont hold the asset anymore, we can remove it
+        if (_nftCollection[GUILD][nftAddr].length() == 0) {
+            _collectedNFTs.remove(nftAddr);
+        }
+    }
+
+    /**
+     * @notice Registers a potential new NFT in the NFT extension.
+     * @notice The caller must have the ACL Flag: REGISTER_NFT.
+     * @dev Reverts if the token address is reserved.
+     * @param nftAddr The address of the new NFT.
+     */
+    function registerPotentialNewNFT(address nftAddr)
+        public
+        hasExtensionAccess(this, AclFlag.REGISTER_NFT)
+    {
+        require(isNotReservedAddress(nftAddr), "reservedToken");
+
+        if (!availableNFTs[nftAddr]) {
+            availableNFTs[nftAddr] = true;
+        }
+    }
+
+    /**
+     * @notice Checks if a given token address is reserved.
+     */
+    function isNotReservedAddress(address addr) public pure returns (bool) {
+        return addr != GUILD && addr != TOTAL && addr != SHARES;
```

Comment:
we should put this logic in a centralized place to make sure when we change it somewhere we change it for everyone

Reply:
yeah, I moved that to the DaoConstant contract.